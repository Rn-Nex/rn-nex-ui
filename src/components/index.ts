export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
