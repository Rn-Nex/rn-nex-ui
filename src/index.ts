export * from './components';
export * from './libraries';
export * from './utils';
