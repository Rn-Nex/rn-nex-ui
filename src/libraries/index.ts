export * from './themes/v2';
export * from './style';
