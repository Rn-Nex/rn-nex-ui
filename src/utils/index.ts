export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
