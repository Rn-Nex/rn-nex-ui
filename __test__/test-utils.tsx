import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
