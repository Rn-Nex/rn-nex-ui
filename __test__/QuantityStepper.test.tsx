import React from 'react';
import { QuantityStepper } from '../src';
import { fireEvent, render } from './test-utils';

describe('QuantityStepper Component', () => {
  const mockOnIncrementTestId = 'on-increment-test-id';
  const mockOnDecrementTestId = 'on-decrement-test-id';

  const mockOnIncrement = jest.fn();
  const mockOnDecrement = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<QuantityStepper value={10} />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should apply the label styles when passed the label props', () => {
    const { getByText } = render(<QuantityStepper value={10} labelProps={{ style: { color: 'red' } }} />);
    const label = getByText('10');
    expect(label.props.style).toEqual(expect.objectContaining({ color: 'red' }));
  });

  it('should trigger the onIncrement function when clicked on the increment button', () => {
    const { getByTestId } = render(
      <QuantityStepper
        onIncrement={mockOnIncrement}
        value={0}
        allowInfiniteIncrement
        onIncrementTestId={mockOnIncrementTestId}
      />,
    );

    const increment = getByTestId(mockOnIncrementTestId);
    fireEvent.press(increment);
    expect(mockOnIncrement).toHaveBeenCalled();
    expect(mockOnIncrement).toHaveBeenCalledTimes(1);
  });

  it('should trigger the onDecrement function when clicked on the decrement button', () => {
    const { getByTestId } = render(
      <QuantityStepper
        onDecrement={mockOnDecrement}
        value={0}
        allowInfiniteDecrement
        onDecrementTestId={mockOnDecrementTestId}
      />,
    );

    const decrement = getByTestId(mockOnDecrementTestId);
    fireEvent.press(decrement);
    expect(mockOnDecrement).toHaveBeenCalled();
    expect(mockOnDecrement).toHaveBeenCalledTimes(1);
  });

  it('should disable the decrement button when passed the disabledDecrement prop', () => {
    const { getByTestId } = render(
      <QuantityStepper value={10} onDecrement={mockOnDecrement} disabledDecrement onDecrementTestId={mockOnDecrementTestId} />,
    );

    const decrement = getByTestId(mockOnDecrementTestId);
    fireEvent.press(decrement);
    expect(mockOnDecrement).not.toHaveBeenCalled();
    expect(mockOnDecrement).toHaveBeenCalledTimes(0);
  });

  it('should disable the increment button when passed the disabledIncrement prop', () => {
    const { getByTestId } = render(
      <QuantityStepper value={2} onIncrement={mockOnIncrement} disabledIncrement onIncrementTestId={mockOnIncrementTestId} />,
    );

    const increment = getByTestId(mockOnIncrementTestId);
    fireEvent.press(increment);
    expect(mockOnDecrement).not.toHaveBeenCalled();
    expect(mockOnIncrement).toHaveBeenCalledTimes(0);
  });
});
