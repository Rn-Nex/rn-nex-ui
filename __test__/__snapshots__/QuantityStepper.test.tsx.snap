// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityStepper Component should render correctly 1`] = `[Function]`;
